uint16 blockid
int64 x
int64 y
int64 z
